src/App.tsx

```diff
@@ -16,46 +16,76 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [type, setType] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const setList = (goods : string[]) => {
+    const visibleGoods = [...goods];
+
+    switch (type) {
+      case 'alphabet':
+        visibleGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case 'length':
+        visibleGoods.sort((a, b) => a.length - b.length);
+        break;
+      case 'reset':
+        return [...goods];
+      default:
+        break;
+    }
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return visibleGoods;
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={`button is-info ${type === 'alphabet' ? '' : 'is-light'}`}
+          onClick={() => setType('alphabet')}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={`button is-success ${type === 'length' ? '' : 'is-light'}`}
+          onClick={() => setType('length')}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={`button is-warning ${isReversed ? '' : 'is-light'}`}
+          onClick={() => setIsReversed(!isReversed)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(isReversed || type) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setIsReversed(false);
+              setType('');
+            }}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {setList(goodsFromServer).map(good => <li data-cy="Good">{good}</li>)}
```

Comment:
Move the result of this function `setList(goodsFromServer)` to a variable and use it.  You can just use useMemo by the way. 

Don't forget to add a key attribute to the list items.
```suggestion
          {goods.map(good => (
            <li data-cy="Good" key={good}>{good}</li>
          ))}
```